sockeye/model.py

```diff
@@ -91,33 +91,36 @@ class SockeyeModel:
     time.
 
     :param config: Model configuration.
+    :param prefix: Name prefix for all parameters of this model.
     """
 
-    def __init__(self, config: ModelConfig) -> None:
+    def __init__(self, config: ModelConfig, prefix: str = '') -> None:
         self.config = copy.deepcopy(config)
         self.config.freeze()
+        self.prefix = prefix
         logger.info("%s", self.config)
 
         # encoder & decoder first (to know the decoder depth)
-        self.encoder = encoder.get_encoder(self.config.config_encoder)
-        self.decoder = decoder.get_decoder(self.config.config_decoder)
+        self.encoder = encoder.get_encoder(self.config.config_encoder, prefix=self.prefix)
+        self.decoder = decoder.get_decoder(self.config.config_decoder, prefix=self.prefix)
 
         # source & target embeddings
-        embed_weight_source, embed_weight_target, out_weight_target = self._get_embed_weights()
+        embed_weight_source, embed_weight_target, out_weight_target = self._get_embed_weights(prefix)
```

Comment:
Nit: Before you used 'self.prefix' and here and the lines below just 'prefix'. I would prefer using self.prefix overall for consistency.